packages/getters/src/compact-block.ts

```diff
@@ -0,0 +1,4 @@
+import { GasPrices } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/fee/v1/fee_pb.js';
+import { createGetter } from './utils/create-getter.js';
+
+export const getAssetIdFromGasPrices = createGetter((gp?: GasPrices) => gp?.assetId);
```

Comment:
If `getAssetIdFromGasPrices` is called with a `GasPrices` object where assetId is undefined, it will still return `undefined`.

Reply:
hmm. this solved the case on the testnet i was using but it's true the inner could be empty.

true correctness presumably requires much idb refactor and some helper methods.

i will add something here but larger interface work is out of scope

Reply:
okay - if an AssetId class instance is present, it always contains a uint8array inner. so this

1. guards against the condition of an absent assetid (no inner, invalid keypath)
2. does not guard against the condition of a present assetid with default zero-length inner (bad inner, valid keypath)
3. does not guard against the impossible condition of a present assetid with an absent inner (can't happen by getter input type)

in the case of 3, the parameter of the getter requires a message class instance, and a class instance of gas prices will never have a present assetid with an absent inner. this is because children of the class instance are always other class instances, and so `assetId` may be absent and thus inner is undefined (case 1), or present and thus inner is defined of some length (case 2)

so the getter performs its purpose, but does not perform length verification. thus 1639